test/ec_internal_test.c

```diff
@@ -259,6 +259,86 @@ static int underflow_test(void)
 }
 #endif
 
+/*
+ * Tests behavior of the decoded_from_explicit_params flag and API
+ */
+static int decoded_flag_test(void)
+{
+    EC_GROUP *grp;
+    EC_GROUP *grp_copy = NULL;
+    ECPARAMETERS *ecparams = NULL;
+    ECPKPARAMETERS *ecpkparams = NULL;
+    unsigned char *encodedparams = NULL;
+    const unsigned char *encp;
+    int encodedlen;
+    int testresult = 0;
+
+    grp = EC_GROUP_new(EC_GFp_simple_method());
+    if (!TEST_ptr(grp)
+        || !TEST_int_eq(grp->decoded_from_explicit_params, 0))
```

Comment:
Should we have an accessor for EC_GROUP as well and test that instead?

Reply:
We should also have tests for the behavior of the EC_KEY_decoded_from_explicit_params wrapper.

Reply:
No, I would like to avoid adding more API calls and the users are simplified by the existing EC_KEY wrapper. IMO the existing accessor should be sufficient. I'll add the test for the wrapper.

Reply:
Ok about not adding the extra accessor to the public API if you don't think it is needed beyond testing.
This matter i consider resolved.

What is unresolved is the fact that we test the internal behavior of the flag inside EC_GROUP but we are missing tests for the public API to ensure it conforms to our documentation.

Reply:
But I added the EC_KEY_decoded_from_explicit_params() to the test code. Is there something else missing?

Reply:
Yes... That I can't read anymore! 🤦‍♂️

Sorry for the noise!